### PR TITLE
Luhn's Algorithm in C

### LINKED ISSUE #2392: Luhn's Algorithm [C/C++]
## 🚀 Feature

Luhn's Algorithm in CPP language & C language.

## Assignment
[C++] & [C]
Hello maintainers!  
As a participant/CA for GSSOC '21,  
I wish to contribute & resolve this issue with proper indentation, comments & sample I/O in C++.
Kindly assign me these issues under GSSOC '21...

## Pitch
This simple algorithm validates credit cards.

### PR DESCRIPTION
**Pull request by Suvraneel**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I've properly studied the contribution guidelines.

### Description

Luhn's Algorithm for credit card validation in C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.
- [x] This is in response to an issue assigned to me under GSSoC'21

## Related Issues or Pull Requests

closes #2392

### CODE REVIEW

**Comment by Suvraneel** [comment]
Hello maintainers!
Kindly review this please....
@HarshCasper @iamrajiv 
@bhav09 @siddharth25pandey

**Review by siddharth25pandey — APPROVED** [review]
@Suvraneel Good to go. Able to merge :)

**Review by iamrajiv — APPROVED** [review]

File: C/cp/LuhnsAlgo.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    char card[16];                               //character array
+    printf("Enter the credit card number:\n");
+    gets (card);                                 //reads a line from stdin and stores it into the 'card' character array
+    int checksum = 0;                            //initiates checksum to zero
+    for (int i=0; i<strlen(card) ; i++)          //iterates as many times as there are characters in the input string
+    {
+        int digit = card[i] - '0';
+        if (i%2==1){                             //even positions
+        checksum += digit;                       //digits added to checksum as it is
+        printf(" + %d", digit);
```
Comment (HarshCasper) [inline C/cp/LuhnsAlgo.c]: Put the Comments above the Line of Code

File: C/cp/LuhnsAlgo.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        if (i%2==0){                             //odd positions (might require 2digit adjustments)
```
Comment (HarshCasper) [inline C/cp/LuhnsAlgo.c]: Put the Comments above the Line of Code

File: C/cp/LuhnsAlgo.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            */
+            printf(" + %d", (2*digit<10) ? (2*digit):(((2*digit)%10)+1));        
+        } 
+    }
+    printf("\t = %d\t",checksum);                //test if checksum is valid (ie, if modulo10 = 0)
+    printf(checksum%10==0 && (int)strlen(card) == 16 ? "(Valid)":"(Invalid)");       
+    if ((int)strlen(card) != 16)                 //the card number must have exactly 16 digits   
+        printf("\n[Invalid because card number must have exactly 16 digits.]");            
+    return 0; 
+}
```
Comment (HarshCasper) [inline C/cp/LuhnsAlgo.c]: Put the Comments above the Line of Code

File: C/cp/LuhnsAlgo.c [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+Checksum = 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 5 = 70
+
+Since the checksum is divisible by ten, 5987654024681355 is a valid credit card number. Write a program
+that, upon input a 16-digit credit card number, determines whether the card number is valid. A 16-digit
+integer cannot be accommodated in a long int variable, so read the input as a string (a character array).
+*******************************************************************************/
+#include <stdio.h>
+#include <string.h>                              //header for string operations
+
+int main()
```
Comment (HarshCasper) [inline C/cp/LuhnsAlgo.c]: Push all the code to a modular function. Rest main() should be only for driver code.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C/cp/LuhnsAlgo.c (hunk 1 above)
Comment (Suvraneel) [reply C/cp/LuhnsAlgo.c]: The inline codes right? 
Sure... right away.. Thanks

**Comment by Suvraneel** [comment]
@HarshCasper 
Resolved all the requested changes :)

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**